cloud/linode/instances.go

```diff
@@ -79,12 +79,27 @@ func (e instanceNoIPAddressesError) Error() string {
 	return fmt.Sprintf("instance %d has no IP addresses", e.id)
 }
 
-func (i *instances) linodeByName(nodeName types.NodeName) (*linodego.Instance, error) {
+func (i *instances) linodeByIP(kNode *v1.Node) (*linodego.Instance, error) {
 	i.nodeCache.RLock()
 	defer i.nodeCache.RUnlock()
+	var kNodeAddresses []string
+	for _, address := range (kNode).Status.Addresses {
+		if address.Type == "ExternalIP" || address.Type == "InternalIP" {
+			kNodeAddresses = append(kNodeAddresses, address.Address)
+		}
+	}
+	if kNodeAddresses == nil {
+		return nil, fmt.Errorf("no IP address found on node %s", kNode.Name)
+	}
 	for _, node := range i.nodeCache.nodes {
-		if node.Label == string(nodeName) {
-			return node, nil
+		var nodeAddresses []string
+		for _, nodeIP := range node.IPv4 {
+			nodeAddresses = append(nodeAddresses, nodeIP.String())
+		}
+		for _, nodeIP := range nodeAddresses {
+			if slices.Contains(kNodeAddresses, nodeIP) {
+				return node, nil
```

Comment:
I was thinking that might be the case, I suspect I'll need to change this to filter out any Linode PrivateIPs

Reply:
I have refactored this logic so we only match on public IP, and only if we don't find a providerID or matching linode label. I understand this will still not work when we fully remove public IPs in VPCs, but this is just meant as a fallback if the other methods do not work